lib/OpenQA/Script/CloneJob.pm

```diff
@@ -226,11 +226,32 @@ sub handle_tx ($tx, $url_handler, $options, $jobs) {
     }
 }
 
+# append a formatted "-NN" to the TEST parameter
+sub append_sequential_number_to_test_name ($n, $post_params) {
+    my $number_suffix = sprintf("-%02d", $n);
+    foreach my $job (keys %$post_params) {
+        my $test_name = $post_params->{$job}->{'TEST'};
+        if ($n == 1) {
+            $test_name .= $number_suffix;    # just append at the first time
+        }
+        else {
+            # from the second onwards, replace the suffix
+            substr($test_name, length($test_name) - length($number_suffix)) = $number_suffix;
+        }
+        $post_params->{$job}->{'TEST'} = $test_name;
+    }
+}
+
 sub clone_jobs ($jobid, $options) {
     my $url_handler = create_url_handler($options);
+    my $how_many_runs = $options->{'repeat'};
+    delete $options->{'repeat'};    # now it's useless, we can remove it
```

Comment:
`$how_many_runs` is a very cumbersome variable name. You can also exploit the return value of `delete`. What about naming it just in accordance with the CLI option?

```suggestion
    my $repeat = delete $options->{'repeat'};
```